code/implementation/host-api/src/main/java/io/cattle/platform/host/service/impl/HostApiServiceImpl.java

```diff
@@ -66,7 +68,13 @@ public HostApiAccess getAccess(Long hostId, Map<String,Object> data) {
 
     protected String getToken(Host host, Map<String,Object> inputData) {
         Map<String,Object> data = new HashMap<String,Object>(inputData);
-        data.put(HOST_UUID, host.getUuid());
+        String uuid = DataAccessor.fields(host).withKey(HostConstants.FIELD_REPORTED_UUID).as(String.class);
+        if(uuid != null){
+            data.put(HostConstants.FIELD_REPORTED_UUID,uuid);
+        }
+        else{
+            data.put(HOST_UUID, host.getUuid());
+        }
```

Comment:
Oh, i see. This is the reason, I had to put that check on ReportedUuid and HostUuid in GO code.I'll change this to HOST_UUID and modify the GO code accordingly.